### PR TITLE
188 review sonarcloud hotspots and resolve all of them

### PR DESCRIPTION
**Pull request by LukeParky**
DESCRIPTION OF PR:

Fixes the majority of security hotspots identified by sonarcloud.
These can be found here: https://sonarcloud.io/project/security_hotspots?id=GeospatialResearch_Digital-Twins&sinceLeakPeriod=true

Closes: #188

Blocked by: #143 - Some of the rec_data code was malfunctioning on windows but it's easier to resolve it in the way done in #143 than double up the work.

Remaining work:
- [X] Solve root geoserver issue, or escalate to another issue (#190)
- [X] Solve root nginx issue, or escalate to another issue (#193)
- [x] Test docker config carefully
  - [x] docker-compose.yml
  - [x] docker-compose-dev.yml
  - [x] docker-compose-dev.yml with windows host python

## Developer Checklist
- [x] Make code change
- [ ] Update tests
  - [ ] Update / create new tests
  - [ ] Ensure these tests have the expected behaviour
  - [ ] Test locally and ensure tests are passing
- [ ] Update documentation
  - [ ] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

## Reviewer Checklist
- [ ] Check new code for code smells
- [ ] Check new tests
  - [ ] Ensure adequate coverage
  - [ ] Check for code smells within tests
- [ ] Check if documentation needs updating
  - [ ] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -46,14 +46,14 @@ Pin-Priority: 1000 \n\
...
 # Install geckodriver, webdriver for firefox, needed for selenium
- && curl --proto "=https" -L https://github.com/mozilla/geckodriver/releases/download/v0.30.0/geckodriver-v0.30.0-linux64.tar.gz | tar xz -C /usr/local/bin \
+ && curl --proto "=https" -L "https://github.com/mozilla/geckodriver/releases/download/v0.30.0/geckodriver-v0.30.0-linux64.tar.gz" | tar xz -C /usr/local/bin \
 # Install health-checker tool that allows us to run commands when checking root endpoint to check if service is available
- && wget -q https://github.com/gruntwork-io/health-checker/releases/download/v0.0.8/health-checker_linux_amd64 -O /usr/local/bin/health-checker \
+ && curl --proto "=https" -L "https://github.com/gruntwork-io/health-checker/releases/download/v0.0.8/health-checker_linux_amd64" -o /usr/local/bin/health-checker \
  && chmod +x /usr/local/bin/health-checker \
 # Cleanup image and remove junk
  && rm -fr /var/lib/apt/lists/* \
 # Remove unused packages. Keep curl for health checking in docker-compose
```
Comment (LukeParky) [inline Dockerfile:55]:
```suggestion
# Remove unused packages.
```

**Review by LukeParky — COMMENTED** [review]
Self-review LGTM

File: Dockerfile (hunk 1 above)
Comment (LukeParky) [reply Dockerfile:55]: We do need curl, do not remove this

**Comment by LukeParky** [comment]
Note of potential error to resolve:

```python
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/app/src/dynamic_boundary_conditions/river/river_data_from_niwa.py", line 234, in fetch_rec_data_from_niwa\n    raise RuntimeError("Failed to fetch \'rec_data\'.")\nRuntimeError: Failed to fetch \'rec_data\'.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 451, in trace_task\n    R = retval = fun(*args, **kwargs)\n                 ^^^^^^^^^^^^^^^^^^^^\n  File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 734, in __protected_call__\n    return self.run(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/app/src/tasks.py", line 184, in generate_river_inputs\n    main_river.main(selected_polygon, **parameters)\n  File "/app/src/dynamic_boundary_conditions/river/main_river.py", line 177, in main\n    river_data_to_from_db.store_rec_data_to_db(engine)\n  File "/app/src/dynamic_boundary_conditions/river/river_data_to_from_db.py", line 84, in store_rec_data_to_db\n    rec_data = load_backup_rec_data_from_niwa()\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/app/src/dynamic_boundary_conditions/river/river_data_to_from_db.py", line 43, in load_backup_rec_data_from_niwa\n    raise FileNotFoundError(f"\'rec_data\' file not found: {rec_data_dir}")\nFileNotFoundError: \'rec_data\' file not found: /stored_data/rec_data\n', 'args': "['POLYGON ((172.71678302522903 -43.400136655560765, 172.71678302522903 -43.370613130921434, 172.65156000179044 -43.370613130921434, 172.65156000179044 -43.400136655560765, 172.71678302522903 -43.400136655560765))']", 'kwargs': '{}', 'description': 'raised unexpected', 'internal': False}
```

**Comment by LukeParky** [comment]
> Note of potential error to resolve:
> 
> ```python
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/app/src/dynamic_boundary_conditions/river/river_data_from_niwa.py", line 234, in fetch_rec_data_from_niwa\n    raise RuntimeError("Failed to fetch \'rec_data\'.")\nRuntimeError: Failed to fetch \'rec_data\'.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 451, in trace_task\n    R = retval = fun(*args, **kwargs)\n                 ^^^^^^^^^^^^^^^^^^^^\n  File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 734, in __protected_call__\n    return self.run(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/app/src/tasks.py", line 184, in generate_river_inputs\n    main_river.main(selected_polygon, **parameters)\n  File "/app/src/dynamic_boundary_conditions/river/main_river.py", line 177, in main\n    river_data_to_from_db.store_rec_data_to_db(engine)\n  File "/app/src/dynamic_boundary_conditions/river/river_data_to_from_db.py", line 84, in store_rec_data_to_db\n    rec_data = load_backup_rec_data_from_niwa()\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/app/src/dynamic_boundary_conditions/river/river_data_to_from_db.py", line 43, in load_backup_rec_data_from_niwa\n    raise FileNotFoundError(f"\'rec_data\' file not found: {rec_data_dir}")\nFileNotFoundError: \'rec_data\' file not found: /stored_data/rec_data\n', 'args': "['POLYGON ((172.71678302522903 -43.400136655560765, 172.71678302522903 -43.370613130921434, 172.65156000179044 -43.370613130921434, 172.65156000179044 -43.400136655560765, 172.71678302522903 -43.400136655560765))']", 'kwargs': '{}', 'description': 'raised unexpected', 'internal': False}
> ``

This will be resolved in #143 so blocked by #143

File: Dockerfile [hunk 2]
```diff
@@ -46,31 +46,30 @@ Pin-Priority: 1000 \n\
...
     && setfacl -R -m u:nonroot:rwx /stored_data \
-    && mkdir /stored_data/geoserver \
-    && mkdir /stored_data/model_output
-
-USER nonroot
+    && mkdir /stored_data/geoserver
 
 # Copy python virtual environment from build layer
-COPY --chown=nonroot:nonroot --chmod=544 --from=build /venv /venv
+COPY --chmod=555 --from=build /venv /venv
```
Comment (LukeParky) [inline Dockerfile:65]: Being more explicit with ownership could be better, Casey mentioned implicit was confusing

```suggestion
COPY --chown=root:root --chmod=555 --from=build /venv /venv
```

File: Dockerfile [hunk 3]
```diff
@@ -46,31 +46,30 @@ Pin-Priority: 1000 \n\
...
+
```
Comment (caseyli92) [inline Dockerfile:66]: As we discussed, let's add the `chown` command to be more explicit.
```suggestion
COPY --chown=root:root --chmod=555 --from=build /venv /venv
```

**Review by caseyli92 — CHANGES_REQUESTED** [review]
Looks good. Just a reminder to address the two new SonarCloud issues.

**Review by caseyli92 — APPROVED** [review]
Looking good :)

**Pull request merged**